### PR TITLE
Make _store tracked, add assertions to warn about disconnected store

### PR DESCRIPTION
**Pull request by rwwagner90**
This is likely not how we want to fix this, but it does help my use case.

### CODE REVIEW

File: addon/-private/model.ts [hunk 1]
```diff
@@ -197,11 +198,11 @@ export default class Model {
   }
 
   private get store(): Store {
```
Comment (dgeb) [inline addon/-private/model.ts:200]:
```suggestion
  protected get store(): Store {
```

File: addon/-private/fields/key.ts [hunk 2]
```diff
@@ -9,7 +9,9 @@ export default function key(options?: KeyDefinition);
 export default function key(options: Model | KeyDefinition = {}, _?: unknown) {
   function trackedKey(target: any, property: string, _: PropertyDescriptor) {
     function get(this: Model) {
-      return getKeyCache(this, property).value;
```
Comment (dgeb) [inline addon/-private/fields/key.ts:12]: Can you add an assertion here as well please?

**Review by dgeb — CHANGES_REQUESTED** [review]
Looks good overall, just needs a couple tweaks. Thanks for adding the assertions.

File: addon/-private/fields/key.ts (hunk 2 above)
Comment (rwwagner90) [reply addon/-private/fields/key.ts:12]: Sure thing, sorry about that. Added!

**Comment by dgeb** [comment]
@rwwagner90 looks good, thanks!

**Pull request merged**